### ISSUE TITLE
Measurement Quick Entry Modal

### ISSUE THREAD

**Issue by ellennelson10** [StraboSpot/StraboField#423]
Here are some mockups for the quick entry modal for measurements.

- For IOS, we attempted to clean up the instructions "tap the compass to take a measurement" (instead of the red text above). Will you need the artwork for the compass with text? The blue text above the compass should take you to manual entry if clicked and then the manual entry form for IOS will just be the same as the Android form.
- Also in IOS, the data in the form should be saved to the notebook but not reset when a measurement is made with the compass. So if a user fills out the fields to say foliation, cleavage, N Side Up in the Quick Entry Modal, each consecutive measurement would also have this data unless they change a field between measurements. If they close the QEM those fields are not saved, but there are still templates if a user did want to save those fields.

![MeasurementsQE-01](https://user-images.githubusercontent.com/91563766/161280139-76ab969c-9c2d-43fb-b154-612324dcd984.png)
![MeasurementsQE-02](https://user-images.githubusercontent.com/91563766/161280146-96bb655e-e4df-4f9e-8bb3-4440661f7742.png)

**Comment by NathanNovak** [comment]
Those look really nice!

**Comment by jessicagood** [comment]
@ellennelson10 Do you want to be able to enter a Tabular Zone from the QEM? You cannot in the current version of SS but you can add a template for TZ. And if someone has a TZ template should that show up under Plane or not at all?

**Comment by ellennelson10** [comment]
@jessicagood No, I don't think that needs to be part of the QEM. If someone has a TZ template have it show up under Plane.

**Comment by jessicagood** [comment]
@ellennelson10 Can you add the new artwork for the compass here which includes the text when you get a chance? Thanks

**Comment by ellennelson10** [comment]
@jessicagood Here is the artwork for the compass. I exported it with a transparent background, but let me know if it needs a white element behind it.

![CompassArt-03](https://user-images.githubusercontent.com/91563766/164111945-8112f49f-12ca-4d7d-b81a-ed0263594bd9.png)

**Comment by jessicagood** [comment]
Added with commits above. This will need rigorous testing as there are so many options, ios/android/, notebook entry/shortcut entry, 0 templates linear/planar, planar + associated linear, 1 planar template, 1 linear template, multiple planar, multiple linear, no planar template + 1 associated linear template, multiple planar templates + associated linear template, multiple planar + 1 associated linear template, etc. Each of the combinations with manual vs compass measurements. They're all handled slightly differently to create the data structure properly.

**Comment by NathanNovak** [comment]
Noticed that adding an associated measurement from the detail page is not handled correctly. It doesn't add it as an associated measurement.

**Comment by jessicagood** [comment]
Ready for testing with commit 7ddcaa024f9b7974ee7edfbe3a38809cc3795daa. Fixes are already in latest testing version of the app.

**Comment by jessicagood** [comment]
I don't think there is anything more to mockup here. It just needs more testing as described in the comment above from Apr 21. @ellennelson10

**Comment by ellennelson10** [comment]
I've tested this quite a bit on iOS and haven't found any major issues. 

There is an odd feature with templates. Say I have a template defined with planar feature type "foliation" and then foliation type of "cleavage". If I change the planar feature type to say "Bedding" the foliation type button disappears (as it should), but when I save the measurement it is saved as "Bedding - CLEAVAGE". This seems like it has to do with how templates operate and is a pretty low priority issue.

**Comment by jessicagood** [comment]
Would you be able to make a video of this issue? I've tried following your steps and variations of them but I haven't been able to recreate the issue yet.

**Comment by jessicagood** [comment]
Thanks @ellennelson10. That helps. It looks like a bug to me but is not an issue on Android (which use the manual measurements mode). If you use manual mode on iOS do you get the same issue?

**Comment by ellennelson10** [comment]
@jessicagood It works like it should with manual measurement input.
